flatfs.go

```diff
@@ -756,43 +797,54 @@ func (fs *Datastore) checkpointDiskUsage(newDuInt int64) {
 	}
 }
 
+// persistDiskUsageFile updates the diskusage file with the last known
+// value
 func (fs *Datastore) persistDiskUsageFile() {
-	// Store DiskUsage on clean shutdowns.
-	du, err := fs.DiskUsage()
-	if err != nil {
-		// do not store on errors. just ignore them
-		return
-	}
+	fs.updateLock.Lock()
```

Comment:
You know, really, we should probably just do this in a goroutine and signal the goroutine with a channel. We could end up with a lot of blocking here, especially because this the "should we update" check is racy. (old bug, I guess, but we should still fix it).

Reply:
Your right, and that was going to be my next step.